OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp

```diff
@@ -143,7 +143,10 @@ runInverseKinematicsWithOrientationsFromFile(Model& model,
 
     TimeSeriesTable_<SimTK::Rotation> orientationsData =
         OpenSenseUtilities::convertQuaternionsToRotations(quatTable);
-
+    
+    // Trim orientation data based on User input times 
+    orientationsData.trim(get_time_range(0),get_time_range(1));
```

Comment:
May want to check that `orientationsData` is non-empty after trim. It doesn't look like [`trim()`](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/TimeSeriesTable.h#L440) guarantees this as a post condition. Also, `trim()` calls [`trimToIndices()`](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/TimeSeriesTable.h#L492), which notes that "Side effect may include that headers/metaData may be left stale." Not sure whether any of the code below assumes the metadata is correct.

Reply:
Thanks @tkuchida I agree that having an empty table is likely due to error so I'd leave this change here as is and modify the trim method to throw on empty table so the check is not repeated elsewhere, I'll make the change to this branch to be part of the PR. Typically meta data does not contain info that depends on number of rows and this is the case here.

Reply:
Ready for review @tkuchida Build failures are due to refactoring on the destination branch so unrelated to this PR

Reply:
Perhaps add test cases to exercise the failure modes and ensure these methods remain well-behaved in the future? I don't know where the tests are for OpenSense's InverseKinematicsStudy, though, as there's no Test directory under OpenSim/Simulation/OpenSense 🤔

Reply:
These are lumped into the testOpensense test case, as we refactor we should break this into more modular tests.